org.scala-ide.sdt.core/src/org/scalaide/ui/editor/DefaultScalaEditorConfiguration.scala

```diff
@@ -0,0 +1,104 @@
+package org.scalaide.ui.editor
+
+import org.eclipse.jface.text.source.IAnnotationHover
+import org.eclipse.jface.preference.IPreferenceStore
+import org.eclipse.jface.text.reconciler.IReconciler
+import org.eclipse.jface.text.reconciler.MonoReconciler
+import org.eclipse.jface.text.source.DefaultAnnotationHover
+import org.scalaide.core.lexical.ScalaPartitions
+import org.eclipse.jface.text.presentation.PresentationReconciler
+import org.scalaide.core.lexical.ScalaCodeScanners
+import org.eclipse.jface.text.source.SourceViewerConfiguration
+import org.eclipse.jface.text.information.InformationPresenter
+import org.eclipse.jdt.ui.text.IJavaPartitions
+import org.eclipse.jface.text.IDocumentListener
+import org.eclipse.jface.text.source.ISourceViewer
+import org.eclipse.jface.text.IDocument
+import org.scalaide.ui.editor.hover.IScalaHover
+import org.eclipse.jface.text.ITextHover
+import org.scalaide.core.IScalaPlugin
+import org.eclipse.jface.text.DocumentEvent
+import org.eclipse.jface.util.PropertyChangeEvent
+import org.eclipse.jface.text.rules.DefaultDamagerRepairer
+
+/** A default SourceViewerConfiguration class for Scala-based editors.
+ *
+ *  This class can be subclassed. It provides a default configuration including
+ *  reconciliation (errors-as-you-type), hovers and Scala syntax highlighting.
+ *
+ *  The workhorse of this implementation is [[InteractiveCompilationUnit]], which provides
+ *  the mapping between original source and Scala-source.
+ *
+ *  Any method can be overwritten.
+ *
+ *  @note Standard content assist is missing from this configuration, but should be easy to add
+ *        in a subsequent release
+ *
+ *  @since 4.1.0
+ */
+trait DefaultScalaEditorConfiguration extends SourceViewerConfiguration {
```

Comment:
I don't like this implementation right now. We already have 3 editor configurations right now (scala core, worksheet, play2), this would add a new one. We should use this one to unify all of them. Most stuff of the SSVC could be moved to this implementation, which then can be mixed into SSVC. This way we would also update the worksheet+play2 to the behavior of the SSVC.